plugins/git/commits/commits_git_reader.go

```
@@ -0,0 +1,98 @@
+package commits
+
+import (
+	"github.com/chrislusf/gleam/util"
+	git "gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
+)
+
+type CommitsGitReader struct {
+	repositoryID string
+	commits      object.CommitIter
+	refs         map[string]struct{}
+}
+
+func New(r *git.Repository, path string) *CommitsGitReader {
```

This should also return an error and all errors should be handled.